src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -2846,6 +2856,39 @@ private static void FormatTrackedParameters(RequestDelegateFactoryContext factor
         }
     }
 
+    private static string[] SplitAndTrim(string[] values)
+    {
+        if (values == null || values.Length == 0)
+        {
+            return [];
+        }
+
+        var result = new List<string>();
+        Span<Range> parts = stackalloc Range[16]; 
```

Comment:
This doesn't seem correct. If there are more than 16 values in a single header it will miss them. Maybe just revert to the previous code for now. We can look at improving the perf later.

Reply:
Thanks @BrennanConroy, I will revert that, I am going to mark this as draft as I'm still working on this.